### ISSUE TITLE
onDestroy not called after goto

### ISSUE THREAD

**Issue by daveychu** [sveltejs/kit#5434]
### Describe the bug

`onDestroy` not called if the template has a transition based on a `store` variable that gets assigned to after calling `goto`.

In our real scenario it was caused by a programming error, but the cause was hard to track down because of multiple layers of derivations. Not sure if there's ever a valid reason for this situation, but confusing if you make the same mistake.

### Reproduction

1. Clone https://github.com/daveychu/sveltekit-ondestroy-not-called-after-goto
2. `npm run dev -- --open`
3. Click "Go to page 2"
4. "pages pile up one after the other on the bottom as I continue navigating back and forth, with no page ever disappearing" if you continuously press the most bottom button -> https://github.com/sveltejs/kit/issues/4772#issuecomment-1126604199

Expected: Only shows page 2
Actual: Shows both pages (or more if you keep going) and a transition on page 1

### System Info

```shell
System:
  OS: macOS 12.4
  CPU: (10) arm64 Apple M1 Pro
  Memory: 296.48 MB / 16.00 GB
  Shell: 5.8.1 - /bin/zsh
Binaries:
  Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
  npm: 8.3.1 - ~/.nvm/versions/node/v16.14.0/bin/npm
Browsers:
  Chrome: 103.0.5060.114
  Safari: 15.5
npmPackages:
  svelte: ^3.44.0 => 3.49.0
```

### Severity

annoyance

**Comment by fernandolguevara** [comment]
@daveychu a workaround is use the `$navigating` store to stop the animation and detach the node 

```html
<script>
  import { navigating } from "$app/stores";
</script>

{#if !$navigating && $store.show}
  <p transition:fly={{ y: 100, duration: 2000 }}>Peekaboo!</p>
{/if}
```

**Comment by Robak08** [comment]
@daveychu I think You can try local transitions as well. These used to work back in Sapper times.

```svelte
{#if $store.show}
  <p transition:fly|local={{ y: 100, duration: 2000 }}>Peekaboo!</p>
{/if}
```

**Comment by dummdidumm** [comment]
I believe this is a Svelte core issue, and I vaguely remember having commented on a similar issue about a minimum reproducible independent of SvelteKit, but I can't find it right now

**Comment by peterpeterparker** [comment]
Ultimatelly we were able to reproduce the issue with another feature on local development too. It helped us narrow the issue to the use of Svelte transitions.

After some search we spotted this changes in Svelte v4 https://svelte.dev/docs/v4-migration-guide#transitions-are-local-by-default and it seems that by applying `|local` to all our transitions, the issue does not appear anymore.

Therefore there is a good chance that this issue is actually the same as https://github.com/sveltejs/svelte/issues/6686.
